addon/utils/load-relationship.js

```diff
@@ -11,14 +11,15 @@ export default function loadAll(model, relationship, dest, options = {}) {
 
     return model.query(relationship, query).then(results => {
         dest.pushObjects(results.toArray());
-
-        var total = results.meta.pagination.total;
-        var pageSize = results.meta.pagination.per_page;
-        var remaining = total - (page * pageSize);
-        if (remaining > 0) {
-            query.page = page + 1;
-            query['page[size]'] = pageSize;
-            return loadAll(model, relationship, dest, query);
+        if (results.meta) {
+            var total = results.meta.pagination.total;
+            var pageSize = results.meta.pagination.per_page;
```

Comment:
Hrm. It feels like this code is relying on somewhere other than itself to not override the query by accident, rather than using something like `Object.assign` to merge with sensible defaults.

I'm ok with this merging based on the above- it was the behavior before- but for future reference this seems like a habit we want to avoid. This isn't new to this PR, but it happened to show up in the diff.